teuthology/parallel.py

```diff
@@ -73,20 +78,13 @@ def __enter__(self):
         return self
 
     def __exit__(self, type_, value, traceback):
-        self.group.join()
-
         if value is not None:
             return False
 
-        try:
-            # raises if any greenlets exited with an exception
-            for result in self:
-                log.debug('result is %s', repr(result))
-                pass
-        except Exception:
-            # Emit message here because traceback gets stomped when we re-raise
-            log.exception("Exception in parallel execution")
-            raise
+        # raises if any greenlets exited with an exception
```

Comment:
I was misreading this as greenlet.get(), not queue.get().  The way the queuing works, any reason for greenlet termination is available immediately.  So I rescind my request for a comment too; either you understand queues or you don't, and I didn't, but now I do